### PR TITLE
Fix new line issue in chatbot

### PR DESCRIPTION
**Pull request by dawoodkhan82**
## Description
Adds `line_breaks` param to chatbot, dataframe, and markdown components.

Sets `line_breaks: True` for chatbot and dataframe, and `False` for markdown

Closes: #4344

## 🎯 PRs Should Target Issues

Before your create a PR, please check to see if there is [an existing issue](https://github.com/gradio-app/gradio/issues) for this change. If not, please create an issue before you create this PR, unless the fix is very small. 

Not adhering to this guideline will result in the PR being closed. 

## Tests

1. PRs will only be merged if tests pass on CI. To run the tests locally, please set up [your Gradio environment locally](https://github.com/gradio-app/gradio/blob/main/CONTRIBUTING.md) and run the tests: `bash scripts/run_all_tests.sh`

2. You may need to run the linters: `bash scripts/format_backend.sh` and `bash scripts/format_frontend.sh`

### CODE REVIEW

**Comment by dawoodkhan82** [comment]
@abidlabs Can you double check this doesn't break any other edge case?

**Comment by dawoodkhan82** [comment]
Will add a test in a bit

**Comment by abidlabs** [comment]
Interesting, we actually set `breaks: false` explicitly here: 

https://github.com/gradio-app/gradio/blob/b27f7583254165b135bf1496a7d8c489a62ba96f/js/markdown/static/utils.ts#L110

coming in from this PR: https://github.com/gradio-app/gradio/pull/5368

Let me see what's going on

**Comment by abidlabs** [comment]
Commented in the original issue: https://github.com/gradio-app/gradio/issues/4344#issuecomment-1741524070

File: gradio/components/chatbot.py [hunk 1]
```diff
@@ -85,6 +86,7 @@ def __init__(
             sanitize_html: If False, will disable HTML sanitization for chatbot messages. This is not recommended, as it can lead to security vulnerabilities.
             render_markdown: If False, will disable Markdown rendering for chatbot messages.
             bubble_full_width: If False, the chat bubble will fit to the content of the message. If True (default), the chat bubble will be the full width of the component.
+            line_breaks: If True (default), will enable GFM line breaks in chatbot messages. If False, single new lines will be ignored.
```
Comment (abidlabs) [inline gradio/components/chatbot.py:89]:
```suggestion
            line_breaks: If True (default), will enable Github-flavored Markdown line breaks in chatbot messages. If False, single new lines will be ignored. Only applies if `render_markdown` is True.
```

File: gradio/components/dataframe.py [hunk 2]
```diff
@@ -89,6 +90,7 @@ def __init__(
             elem_id: An optional string that is assigned as the id of this component in the HTML DOM. Can be used for targeting CSS styles.
             elem_classes: An optional list of strings that are assigned as the classes of this component in the HTML DOM. Can be used for targeting CSS styles.
             wrap: if True text in table cells will wrap when appropriate, if False the table will scroll horizontally. Defaults to False.
+            line_breaks: If True (default), will enable GFM line breaks in chatbot messages. If False, single new lines will be ignored.
```
Comment (abidlabs) [inline gradio/components/dataframe.py:93]:
```suggestion
            line_breaks: If True (default), will enable Github-flavored Markdown line breaks in chatbot messages. If False, single new lines will be ignored. Only applies for columns of type "markdown." 
```

File: gradio/components/markdown.py [hunk 3]
```diff
@@ -49,12 +50,14 @@ def __init__(
             elem_id: An optional string that is assigned as the id of this component in the HTML DOM. Can be used for targeting CSS styles.
             elem_classes: An optional list of strings that are assigned as the classes of this component in the HTML DOM. Can be used for targeting CSS styles.
             sanitize_html: If False, will disable HTML sanitization when converted from markdown. This is not recommended, as it can lead to security vulnerabilities.
+            line_breaks: If True, will enable GFM line breaks in chatbot messages. If False (default), single new lines will be ignored.
```
Comment (abidlabs) [inline gradio/components/markdown.py:53]:
```suggestion
            line_breaks: If True, will enable Github-flavored Markdown line breaks in chatbot messages. If False (default), single new lines will be ignored.
```

**Review by abidlabs — APPROVED** [review]
Looks great, thanks @dawoodkhan82!
